pkg/proc/native/threads_windows.go

```diff
@@ -158,18 +157,18 @@ func (t *nativeThread) ReadMemory(buf []byte, addr uint64) (int, error) {
 }
 
 func (t *nativeThread) restoreRegisters(savedRegs proc.Registers) error {
-	return _SetThreadContext(t.os.hThread, savedRegs.(*winutil.AMD64Registers).Context)
+	return SetThreadContext(t.os.hThread, savedRegs.(winutil.Registers).Ctx())
```

Comment:
I'd rather have two implementations of this, one for each architecture, than have an implementation of Ctx for each architecture and the winutil.Registers interface.

Reply:
Done! It does make it clear.